test/PrintAsObjC/availability.swift

```diff
@@ -570,27 +595,49 @@ extension Availability {
 }
 
 @objc(SWTReplacementAvailable) class ReplacementAvailable {
-    @objc(replacingMethodInReplacementClassWithFirst:second:) func methodReplacingInReplacementClass(first: Int, second: Int) -> Void {}
-  
+    @objc(replacingMethodInReplacementClassWithPrimitiveParametersWithFirst:second:)
+    func methodReplacingInReplacementClassWithPrimitiveParameters(first: Int, second: Int) -> Void {}
+    
+    @objc(replacingMethodInReplacementClassWithSuperclassObjectParametersWithFirst:second:)
+    func methodReplacingInReplacementClassWithClassObjectParameters(first: SubclassOfReplacementAvailable, second: SubclassOfReplacementAvailable) -> Void {}
```

Comment:
See, it's good to have the test case: this is backwards! You can rename from a method that takes "Subclass" to a method that takes "Base", but not the other way around (because some existing calls might break).

Alternately, maybe we shouldn't be matching types at all, since sometimes people *do* do incompatible renames. What do you think?

Reply:
> I did the `reversed type` parameter test based on the Variance type system.

Ah, but parameters are contravariant, not covariant.

> On the point of not matching types, I agree with you not considering the point you raised. I think The matching type is then extended work of the overloading lookup improvement.
> 
> So should we remove the matching type for now and may consider to implement it when someone asks in the forum?

Yeah, let's go with that. Sorry for sending you down a dead end!

Reply:
> Ah, but parameters are contravariant, not covariant.
Correct, parameters are contravariant thus I implemented the test that the parameters of the renamed to methods have the super type of the parameters of the deprecated method. Or I may misunderstood it and implemented the wrong cases. Could you check this for me again if I implemented it wrong or not

> Yeah, let's go with that. Sorry for sending you down a dead end!
Ok. Let’s go with that and please don’t worry about sending me down a dead end. I also want to go with the best approach for Swift too :)